src/main/java/com/brightpath/learnify/domain/binding/BindingService.java

```diff
@@ -81,8 +81,8 @@ public void removeBindingForNote(UUID noteId) {
         bindingRepository.deleteAllByNoteId(noteId);
     }
 
-    private void checkIfNoteExists(UUID noteId) {
-        noteService.getNoteById(noteId);
+    private void checkIfNoteExists(UUID noteId, String userId) {
```

Comment:
Let's add another method in NoteService. It's weird that we have to pass userId to check if entity exists and userId doesn't matter here